### PR TITLE
[Draft] Introduce `circle-interpreter`

### PR DESCRIPTION
**Pull request by yunjayh**
This commit introduces `circle-interpreter` as driver of `luci-interpreter`.

Signed-off-by: yunjay-hong <yunjay.hong@samsung.com>

---

for #9634

### CODE REVIEW

**Comment by glistening** [comment]
(optional) There is a typo in title: `circle-interperter` should be `circle-interpreter`.

**Comment by yunjayh** [comment]
I'll try to add another tests via `one-infer` to check the arguments validity.

File: compiler/one-cmds/tests/prepare_test_materials.sh [hunk 1]
```diff
@@ -105,6 +105,13 @@ if [[ ! -s "Net_InstanceNorm_003.part" ]]; then
     # https://github.com/Samsung/ONE/issues/8570#issuecomment-1115804257
 fi
 
+if [[ ! -s "Conv2D_000.circle" ]]; then
+    rm -rf Net_InstanceNorm_003.zip
+    wget https://github.com/Samsung/ONE/files/9466423/Conv2D_000.zip
+    unzip Net_InstanceNorm_003.zip
+    # https://github.com/Samsung/ONE/issues/9653#issuecomment-1233668729
+fi
```
Comment (yunjayh) [inline compiler/one-cmds/tests/prepare_test_materials.sh:113]: I uploaded `Conv2D_000.circle` and its output which are generated from `common-artifacts` for test `circle-interpreter` through `one-infer`. Those files are uploaded [here](https://github.com/Samsung/ONE/issues/9653).

File: compiler/one-cmds/tests/prepare_test_materials.sh [hunk 2]
```diff
@@ -105,6 +105,13 @@ if [[ ! -s "Net_InstanceNorm_003.part" ]]; then
     # https://github.com/Samsung/ONE/issues/8570#issuecomment-1115804257
 fi
 
+if [[ ! -s "Conv2D_000.circle" ]]; then
+    rm -rf Conv2D_000.zip
+    wget https://github.com/Samsung/ONE/files/9466423/Conv2D_000.zip
+    unzip Conv2D_000.zip
+    # https://github.com/Samsung/ONE/issues/9653#issuecomment-1233668729
+fi
```
Comment (yunjayh) [inline compiler/one-cmds/tests/prepare_test_materials.sh:113]: I uploaded `Conv2D_000.circle` and its output which are generated from `common-artifacts` for test `circle-interpreter` through `one-infer`. Those files are uploaded [here](https://github.com/Samsung/ONE/issues/9653).

File: compiler/one-cmds/tests/one-infer_neg_006.test [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# limitations under the License.
+
+# invoke circle-interpreter with not-existing circle model
+
+filename_ext="$(basename -- $0)"
+filename="${filename_ext%.*}"
+
+trap_err_onexit()
+{
+  return_code=$?
```
Comment (jinevening) [inline compiler/one-cmds/tests/one-infer_neg_006.test:24]: return_code is not used anywhere

File: compiler/one-cmds/tests/prepare_test_materials.sh [hunk 4]
```diff
@@ -119,7 +126,7 @@ function files_missing() {
 
 declare -a TEST_RECCURENT_MODELS=(\
   "RNN.onnx" "RNN-nobias.onnx" "RNN-relu.onnx" "RNN-bi.onnx" "RNN-noinit.onnx"\
-  "LSTM.onnx" "LSTM-bi.onnx" "LSTM-noinit.onnx" "LSTM-nobias.onnx"
+  "LSTM.onnx" "LSTM-bi.onnx" "LSTM-noinit.onnx" "LSTM-nobias.onnx" "Conv2D_000.circle"
```
Comment (jinevening) [inline compiler/one-cmds/tests/prepare_test_materials.sh:129]: Conv2D_000.circle is reccurent model?

Comment (yunjayh) [reply compiler/one-cmds/tests/prepare_test_materials.sh:129]: 😓 I misread the script. I'll remove it. Thank you for pointing out!

**Pull request closed without merging**